app/scripts/background.js

```diff
@@ -191,33 +196,53 @@ function setupController (initState, initLangCode) {
   //
   // connect to other contexts
   //
-
   extension.runtime.onConnect.addListener(connectRemote)
+
+  const metamaskInternalProcessHash = {
+    [ENVIRONMENT_TYPE_POPUP]: true,
+    [ENVIRONMENT_TYPE_NOTIFICATION]: true,
+    [ENVIRONMENT_TYPE_FULLSCREEN]: true,
+  }
+
+  const isClientOpenStatus = () => {
+    return popupIsOpen || Object.keys(openMetamaskTabsIDs) || notificationIsOpen
```

Comment:
This should be checking `length` on `Object.keys(openMetamaskTabsIDs)`, otherwise `isClientOpenStatus` will return an empty array when it should return `false`. It should be something like:

```js
return popupIsOpen || Object.keys(openMetamaskTabsIDs).length || notificationIsOpen
```